File: problems/tetris/starter/cpp/exampleBot.cpp
#include <iostream>
#include <stdlib.h>
#include <time.h>
#include "Tetris.h"

// A simple example bot which randomly chooses a move.

int main ()
{
    srand ( time(NULL) ); //initialize the random seed
    Board B("input.txt");

    const char possibleMoves[4] = {'L', 'R', 'X', ' '};
    int RandIndex;
    try {
        while (true) {
            RandIndex = rand() % 4;
            B.makeMove(possibleMoves[RandIndex]);
        }
    } catch (std::runtime_error e) {
        std::cout << "Error: " << e.what() << std::endl;
    }

    B.outputMovesToFile("output.txt");
    std::cout << "Output moves to file 'output.txt'" << std::endl;
    return 0;
}

File: problems/tetris/starter/python/exampleBot.py
import Tetris
from random import choice

possibleMoves = [' ', 'L', 'R', 'X']

# A simple example bot that moves randomly
filename = input('Board filename: ')
B = Tetris.Board(filename)
try:
	while True:
		B.makeMove(choice(possibleMoves))
except Exception as e:
	pass
B.outputMovesToFile('output.txt')
print('Output moves to output.txt')
